release/7-3/mariner2-arm64/docker/Dockerfile

```diff
@@ -0,0 +1,73 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+FROM mcr.microsoft.com/cbl-mariner/base/core:2.0 AS installer-env
+
+# Define Args for the needed to add the package
+ARG PS_VERSION=7.3.1
+ARG PS_PACKAGE=powershell-${PS_VERSION}-linux-arm64.tar.gz
+ARG PS_PACKAGE_URL=https://github.com/PowerShell/PowerShell/releases/download/v${PS_VERSION}/${PS_PACKAGE}
+ARG PS_INSTALL_VERSION=7
+
+# Define the folder we will be installing PowerShell to.
+ENV PS_INSTALL_FOLDER=/opt/microsoft/powershell/${PS_INSTALL_VERSION}
+
+# Create the install folder.
+RUN mkdir -p ${PS_INSTALL_FOLDER}
+
+ARG PS_PACKAGE_URL_BASE64
+
+RUN --mount=type=cache,target=/var/cache/tdnf \
+    tdnf update -y \
+    && tdnf install -y ca-certificates wget tar
+
+RUN echo 'in task' \
+    && if [ -n "${PS_PACKAGE_URL_BASE64}" ];  then \
+        echo 'using base64' \
+        && export url=$(echo "${PS_PACKAGE_URL_BASE64}" | base64 --decode -); \
+    else \
+        echo 'using unencoded' \
+        && export url="${PS_PACKAGE_URL}"; \
+    fi \
+    && echo "url: $url" \
+    && wget -O /tmp/powershell.tar.gz "$url" \
+    && echo 'task done'
+
+RUN ls -l /tmp/powershell.tar.gz
+
+# Unzip the linux powershell.tar.gz
+RUN tar zxf /tmp/powershell.tar.gz -C ${PS_INSTALL_FOLDER}
+
+FROM mcr.microsoft.com/cbl-mariner/base/core:2.0 AS final-image
```

Comment:
specify the platform so that we are sure we are getting the correct architecture